src/PythonWrapper.cpp

```diff
@@ -219,47 +229,58 @@ void PySendExceptionCallback(CMQException e, CMessage *msg, void *pyCallback){
+                     (void *) pyCallback);
 }
 
-PySendResult PySendBatchMessage(void *producer, void *batchMessage) {
+
+PySendResult PySendMessageOrderly(void *producer, void *msg, int autoRetryTimes, void *args, PyObject *queueSelector) {
     PySendResult ret;
     CSendResult result;
-    SendBatchMessage((CProducer *) producer, (CBatchMessage *) batchMessage, &result);
+    PyUserData userData = {queueSelector, args};
+    SendMessageOrderly((CProducer *) producer, (CMessage *) msg, &PyOrderlyCallbackInner, &userData, autoRetryTimes,
+                       &result);
     ret.sendStatus = result.sendStatus;
     ret.offset = result.offset;
     strncpy(ret.msgId, result.msgId, MAX_MESSAGE_ID_LENGTH - 1);
     ret.msgId[MAX_MESSAGE_ID_LENGTH - 1] = 0;
     return ret;
 }
 
+int PyOrderlyCallbackInner(int size, CMessage *msg, void *args) {
+    PyUserData *userData = (PyUserData *) args;
+    int index = boost::python::call<int>(userData->pyObject, size, (void *) msg, userData->pData);
+    return index;
+}
 
-PySendResult PySendMessageOrderly(void *producer, void *msg, int autoRetryTimes, void *args, PyObject *queueSelector) {
+PySendResult PySendMessageOrderlyByShardingKey(void *producer, void *msg, const char *shardingKey) {
     PySendResult ret;
     CSendResult result;
-    PyUserData userData = {queueSelector,args};
-    SendMessageOrderly((CProducer *) producer, (CMessage *) msg, &PyOrderlyCallbackInner, &userData, autoRetryTimes, &result);
+    SendMessageOrderlyByShardingKey((CProducer *) producer, (CMessage *) msg, shardingKey, &result);
     ret.sendStatus = result.sendStatus;
     ret.offset = result.offset;
     strncpy(ret.msgId, result.msgId, MAX_MESSAGE_ID_LENGTH - 1);
     ret.msgId[MAX_MESSAGE_ID_LENGTH - 1] = 0;
     return ret;
 }
 
-int PyOrderlyCallbackInner(int size, CMessage *msg, void *args) {
-    PyUserData *userData = (PyUserData *)args;
-    int index = boost::python::call<int>(userData->pyObject, size, (void *) msg, userData->pData);
-    return index;
+CTransactionStatus PyLocalTransactionExecuteCallback(CProducer *producer, CMessage *msg, void *data) {
+    PyUserData *localCallback = (PyUserData *) data;
+    CTransactionStatus status = boost::python::call<CTransactionStatus>(localCallback->pyObject, (void *) msg,
+                                                                        localCallback->pData);
+    return status;
 }
 
-PySendResult PySendMessageOrderlyByShardingKey(void *producer, void *msg, const char *shardingKey) {
+PySendResult PySendMessageInTransaction(void *producer, void *msg, PyObject *localTransactionCallback, void *args) {
+    PyUserData userData = {localTransactionCallback, args};
     PySendResult ret;
     CSendResult result;
-    SendMessageOrderlyByShardingKey((CProducer *) producer, (CMessage *) msg, shardingKey, &result);
+    SendMessageTransaction((CProducer *) producer, (CMessage *) msg, &PyLocalTransactionExecuteCallback, &userData,
```

Comment:
Can't find `SendMessageTransaction` function, not even function prototype.

Reply:
Relative c exporter hasn't released and it will be published ASAP, you can ignore this for now.